pkg/controller/volume/expand/expand_controller.go

```diff
@@ -290,6 +283,18 @@ func (expc *expandController) syncHandler(key string) error {
 		return nil
 	}
 
+	claimClass := v1helper.GetPersistentVolumeClaimClass(pvc)
+	if claimClass == "" {
+		klog.V(4).Infof("volume expansion is disabled for PVC without StorageClasses: %s", util.ClaimToClaimKey(pvc))
+		return nil
+	}
+	class, err := expc.classLister.Get(claimClass)
+	if err != nil {
+		klog.V(4).Infof("failed to expand PVC: %s with error: %v", util.ClaimToClaimKey(pvc), err)
+		return nil
+	}
+
+	volumeResizerName := class.Provisioner
 	return expc.expand(pvc, pv, volumeResizerName)
```

Comment:
spoke with @msau42 offline and I think we don't need to read storageclass at all in expand_controller. Can you just get resizer name from `volumePlugin.GetPluginName()` which should be same as provisioner name for all intree plugin types and for out of tree plugins, we don't really set those here anyways.

Reply:
Nice suggestion indeed, thank you! I now adapted to `volumePlugin.GetPluginName()` and removed the StorageClass lister. You can have another look.